research/quantifying-data-leakage/src/nvflare_gradinv/utils/create_job_config.py

```diff
@@ -0,0 +1,207 @@
+
+    print(
+        f"Created {out_client_config_file} to use DATA_ROOT={client_config['DATA_ROOT']}, DATASET_JSON={client_config['DATASET_JSON']}, and SIGMA0={client_config['SIGMA0']}"
+    )
+    if "PRIOR_FILE" in client_config:
+        print(f"Used prior_file {client_config['PRIOR_FILE']}")
+
+
+def create_gradinv_config(inv_config_file, args, client):
+    inv_config = load_config(inv_config_file)
+
+    inv_config["img_prior"] = args.prior_file
+
+    out_inv_config_file = os.path.join(args.job_config, f"site-{client}", "config", "config_inversion.json")
+    if not os.path.isdir(os.path.dirname(out_inv_config_file)):
+        os.makedirs(os.path.dirname(out_inv_config_file))
+
+    save_config(out_inv_config_file, inv_config)
+
+    print(f"Created {out_inv_config_file} to use prior_file {inv_config['img_prior']}")
+
+
+def create_meta_config(clients, meta_config_file, args):
+    meta_config = load_config(meta_config_file)
+
+    deploy_map = {"server": ["server"]}
+    for client in clients:
+        deploy_map.update({f"site-{client}": [f"site-{client}"]})
+
+    meta_config["deploy_map"] = deploy_map
+
+    out_meta_config_file = os.path.join(args.job_config, "meta.json")
+    if not os.path.isdir(os.path.dirname(out_meta_config_file)):
+        os.makedirs(os.path.dirname(out_meta_config_file))
+
+    save_config(out_meta_config_file, meta_config)
+
+    print(f"Created {out_meta_config_file}")
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--app_folder",
+        type=str,
+        default="app_template",
+        help="Folder containing app config files in JSON format.",
+    )
+    parser.add_argument(
+        "--job_config",
+        type=str,
+        help="Output folder containing the job config.",
+    )
+    parser.add_argument(
+        "--data_root",
+        type=str,
+        help="Data root.",
+    )
+    parser.add_argument(
+        "--dataset_json_prefix",
+        type=str,
+        default="data_200val+rest_client",
+        help="Prefix name for dataset JSON file.",
+    )
+    parser.add_argument(
+        "--n_clients",
+        type=int,
+        default=1,
+        help="Number of clients to generate.",
+    )
+    parser.add_argument(
+        "--num_rounds",
+        type=int,
+        default=100,
+        help="Number of federated learning rounds.",
+    )
+    parser.add_argument(
+        "--invert_clients",
+        type=str,
+        default="9",
+        help="Comma-separated list of client numbers to invert. Defaults to `high-risk` client 9.",
+    )
+    parser.add_argument(
+        "--batch_sizes",
+        type=str,
+        default="1",
+        help="Comma-separated list of batches for each client.",
+    )
+    parser.add_argument(
+        "--sigma0",
+        type=float,
+        default=0.0,
+        help="Noise level for `GaussianPrivacy` filter.",
+    )
+    parser.add_argument("--prior_file", type=str, help="Prior image filename")
```

Comment:
I prefer put "required arguments" in front of "optional arguments"

Reply:
Good point. Changed.

Reply:
thanks for the updates.
I see your "prior_file" is also a required argument, do we want to move it to the front as well?
or is it meant to be an optional argument?

Reply:
Oh. I missed that. Changed.